source/common/stats/tag_extractor_impl.cc

```diff
@@ -114,13 +114,27 @@ bool TagExtractorStdRegexImpl::extractTag(TagExtractionContext& context, std::ve
                                                      regex_) &&
       match.size() > 1) {
     // remove_subexpr is the first submatch. It represents the portion of the string to be removed.
-    const auto& remove_subexpr = match[1];
+    uint32_t idx;
+    for (idx = 1; idx < match.size(); idx++) {
+      if (match[idx].matched) {
+        break;
+      }
+    }
+    const auto& remove_subexpr = match[idx];
```

Comment:
If there is no match then idx==match.size() and this statement will (I believe) access memory outside the bounds of the array.